src/data/polynomial/mirror.lean

```diff
@@ -0,0 +1,185 @@
+/-
+Copyright (c) 2020 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import data.polynomial.ring_division
+import algebra.big_operators.nat_antidiagonal
+
+/-!
+# Reverse of a univariate polynomial
+
+In this file we define `mirror`, a variant of `reverse`. The difference between `reverse` and
+`mirror` is that `reverse` will decrease the degree if the polynomial is divisible by `X`.
+We also define `norm2`, which is the sum of the squares of the coefficients of a polynomial.
+It is also a coefficient of `p * p.mirror`.
+
+## Main definitions
+
+- `p.mirror`
+- `p.norm2`
+
+## Main results
+
+- `mirror_mul_of_domain`: `mirror` preserves multiplication.
+- `irreducible_of_mirror`: an irreducibility criterion involving `mirror`
+- `norm2_eq_mul_reverse_coeff`: `norm2` is a coefficient of `p * p.mirror`
```

Comment:
Some module doc suggestions. Note that only fully-qualified names become links in the online docs:
```suggestion
# "Mirror" of a univariate polynomial

In this file we define `polynomial.mirror`, a variant of `polynomial.reverse`. The difference 
between `reverse` and `mirror` is that `reverse` will decrease the degree if the polynomial is 
divisible by `X`. We also define `polynomial.norm2`, which is the sum of the squares of the 
coefficients of a polynomial. It is also a coefficient of `p * p.mirror`.

## Main definitions

- `polynomial.mirror`
- `polynomial.norm2`

## Main results

- `polynomial.mirror_mul_of_domain`: `mirror` preserves multiplication.
- `polynomial.irreducible_of_mirror`: an irreducibility criterion involving `mirror`
- `polynomial.norm2_eq_mul_reverse_coeff`: `norm2` is a coefficient of `p * p.mirror`
```